### PR TITLE
refactor backport

### PR DESCRIPTION
**Pull request by farodin91**
part of #3247

Signed-off-by: Jan Jansen <jan.jansen@gdata.de>

-----

Thank you for contributing to JanusGraph!

In order to streamline the review of the contribution we ask you
to ensure the following steps have been taken:

### For all changes:
- [ ] Is there an issue associated with this PR? Is it referenced in the commit message?
- [ ] Does your PR body contain #xyz where xyz is the issue number you are trying to resolve?
- [ ] Has your PR been rebased against the latest commit within the target branch (typically `master`)?
- [ ] Is your initial contribution a single, squashed commit?

### For code changes:
- [ ] Have you written and/or updated unit tests to verify your changes?
- [ ] If adding new dependencies to the code, are these dependencies licensed in a way that is compatible for inclusion under [ASF 2.0](https://www.apache.org/legal/resolved.html#category-a)?
- [ ] If applicable, have you updated the LICENSE.txt file, including the main LICENSE.txt file in the root of this repository?
- [ ] If applicable, have you updated the NOTICE.txt file, including the main NOTICE.txt file found in the root of this repository?

### For documentation related changes:
- [ ] Have you ensured that format looks appropriate for the output in which it is rendered?

### CODE REVIEW

File: .backportrc.json [hunk 1]
```diff
@@ -4,6 +4,8 @@
     "targetBranchChoices": ["v0.6"],
     "branchLabelMapping": {
       "^backport/(.+)$": "$1"
-    },    
-    "targetPRLabels": ["backport/skip"]    
+    },
+    "autoMerge": true,
```
Comment (FlorianHockmann) [inline .backportrc.json:8]: Will this wait for all checks to succeed?

File: .github/workflows/backport.yml [hunk 2]
```diff
@@ -16,15 +16,21 @@ on:
   pull_request_target:
     types: ["labeled", "closed"]
 
+permissions:
```
Comment (FlorianHockmann) [inline .github/workflows/backport.yml:19]: How is this working right now without these permissions?

**Review by FlorianHockmann — COMMENTED** [review]
Just saw this PR. I'll need to update the PR where I document the backporting (#3409) after we merge this one.

I think auto-merge makes sense and it's also good to simplify the labels here.
You'll unfortunately have to rebase this as I've already modified the labels Dependabot creates.

File: .backportrc.json (hunk 1 above)
Comment (farodin91) [reply .backportrc.json:8]: https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/incorporating-changes-from-a-pull-request/automatically-merging-a-pull-request

File: .github/workflows/backport.yml (hunk 2 above)
Comment (farodin91) [reply .github/workflows/backport.yml:19]: It was the default without permissions. They are starting to change it. For example updating github workflows isn't possible without this.

**Comment by farodin91** [comment]
@FlorianHockmann updated.

**Review by FlorianHockmann — APPROVED** [review]

**Comment by FlorianHockmann** [comment]
@farodin91 Do you want more people to review this or can we merge it now?

**Comment by farodin91** [comment]
I want it to merge via github app, but i don't have the rights in the app so. i forgot it.

**Pull request merged**